### PR TITLE
Redesigned the Preference page

### PR DESCRIPTION
**Pull request by Izcomayana**
# Fixes Issue/Linear Ticket

( This could be an existing issue or a linear ticket )

**My PR Closes #issue_number_here** or
**My PR Fixes ID-48*

# Changes proposed

## What were you told to do?

The user should have the option to select what appears on his or her sidebar.
Choose either a light or dark theme
Language and region preference
Privacy accessibility: user should decide who sees they are on Zuri App to connect
Notification: user should decide on what notifications to receive

## What did you do?

I made it soo the user should have the option to select what appears on his or her sidebar  
The user can choose either a light or dark theme
Privacy accessibility: user should decide who sees they are on Zuri App to connect
Notification: user should decide on what notifications to receive
I used bootdtrap for the styling

# Check List (Check all the applicable boxes)

🚨Please review the [style guide for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/STYLING.md) and [guidelines for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/CONTRIBUTING.md) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos

image
image
image
image

### CODE REVIEW

**Comment by JulianaSau** [comment]
Fix conflicts

**Comment by JulianaSau** [comment]
Why is the page so dull? Like there should be some green there like the other modals in Zuri Chat.
Black and white are not the only colors that exist🥲🥲🥲

**Review by JulianaSau — CHANGES_REQUESTED** [review]
You've also added so many svgs when we have an icon library called `react-icons` that contains all of these. Please try as much as possible to use available icons and not add all those svgs

**Comment by Izcomayana** [comment]
> Why is the page so dull? Like there should be some green there like the other modals in Zuri Chat. Black and white are not the only colors that exist🥲🥲🥲

File: packages/main/src/App.js [hunk 1]
```diff
@@ -24,8 +24,11 @@ import {
   ResetPassword,
   ChangePassword,
   WhyZuriChat,
-  PrivacyPage
+  WhyZuri,
```
[the remark this answers is not in the document]
Comment (Izcomayana) [reply packages/main/src/App.js:27]: It was a mistake and I have rectify it

**Comment by Izcomayana** [comment]
> Why is the page so dull? Like there should be some green there like the other modals in Zuri Chat. Black and white are not the only colors that exist🥲🥲🥲

That was how the design looked and there are some green in those pages too

**Comment by JulianaSau** [comment]
Resolve conflicts so that i can review again
And please send a figma link to the redesign that your team did. @Izcomayana

File: packages/main/src/pages/preferences/assets/css/main.css [hunk 2]
```diff
@@ -0,0 +1,134 @@
+/* @import { FaStar } from "react-icons/fa"; */
```
Comment (ickynavigator) [inline packages/main/src/pages/preferences/assets/css/main.css:1]: remove comment

File: packages/main/src/pages/preferences/assets/css/main.css [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+input[type="checkbox"]::before {
+  content: "";
+  width: 0.65em;
+  height: 0.65em;
+  border-radius: 50%;
+}
+
+input[type="checkbox"]:checked{
+  background-image: url(../images/checkbox.png);
```
Comment (ickynavigator) [inline packages/main/src/pages/preferences/assets/css/main.css:119]: these images are not necessary
the box can be styled that way

File: packages/main/src/pages/preferences/components/Appearance.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            </div>
+            <div className="d-flex align-items-center mb-3">
+              <label htmlFor="checkbox3" className="position-relative">
+                <input
+                  type="radio"
+                  name="mode"
+                  id="checkbox3"
+                  className="position-absolute bottom-0 end-0"
+                />
+                <img src={darkMode} alt="" className="w-100" />
```
Comment (ickynavigator) [inline packages/main/src/pages/preferences/components/Appearance.js:42]: add a descriptive alt

File: packages/main/src/pages/preferences/components/Appearance.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+          <div className="mt-4 flex-column flex-xl-row d-flex">
+            <div className="d-flex align-items-center mb-3">
+              <label htmlFor="checkbox2" className=" position-relative">
+                <input
+                  type="radio"
+                  name="mode"
+                  id="checkbox2"
+                  className="position-absolute bottom-0 end-0"
+                />
+                <img src={whiteMode} alt="" className="me-5 w-100" />
```
Comment (ickynavigator) [inline packages/main/src/pages/preferences/components/Appearance.js:31]: add a descriptive alt

File: packages/main/src/pages/preferences/components/Notifications.js [hunk 6]
```diff
@@ -0,0 +1,183 @@
+import React from "react";
+import { BsQuestionCircle } from "react-icons/bs";
+import { FaStar } from "react-icons/fa";
+
+import styles from "../assets/css/notifications.css";
+import "../assets/css/main.css";
```
Comment (ickynavigator) [inline packages/main/src/pages/preferences/components/Notifications.js:6]: use modules

File: packages/main/src/pages/preferences/index.js [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+export default function index() {
+  const [showAccessibility, setShowAccessibility] = useState(false);
+  const [showAppearance, setShowAppearance] = useState(false);
+  const [showLangRegion, setShowLangRegion] = useState(false);
+  const [showNotifications, setShowNotifications] = useState(true);
+  const [showPrivacy, setShowPrivacy] = useState(false);
+  const [showSidebar, setShowSidebar] = useState(false);
+  const [preference, setPreference] = useState("notifications");
+
+  const accessibility = () => {
```
Comment (ickynavigator) [inline packages/main/src/pages/preferences/index.js:27]: what happens when someone wants to add a new preference type?
Do they have to add a setState to all the functions?

change this and use a single object to store all the states
you can now loop through the object and change everything to false except the one you want to update

**Review by ickynavigator — CHANGES_REQUESTED** [review]

**Pull request closed without merging**